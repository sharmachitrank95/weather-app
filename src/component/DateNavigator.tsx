import React from 'react';
import {List,ListItem, Grid} from '@material-ui/core'
interface IDateNavigatorProps{
    dates:any,
    currentDate:any,
    handleDateChange:(e:any,value:any)=>void
}
interface IDateNavigatorStates{

}
function getListItemStyle(match:boolean) {
    if(match){
        return {
            backgroundColor: 'grey',
        };
    }
}
export class DateNavigator extends React.Component<IDateNavigatorProps,IDateNavigatorStates>{
    constructor(props:any) {
        super(props);
    }

    render(){
        let dates;
        dates = this.props.dates.map((date:any) => {
               return <ListItem 
               button 
               component={"a"} 
               value={date} 
               onClick={e => this.props.handleDateChange(e,date)}
               style={getListItemStyle(date === this.props.currentDate)}>
               {date}
               </ListItem>
           })
        
        return(
            <Grid item xs={2}>
                <h4 style={{paddingLeft:'1rem'}}>Choose Date:</h4>
                <List component={"nav"}>
                    {dates}
                </List>
            </Grid>
        )
        
    }
}

export default DateNavigator;