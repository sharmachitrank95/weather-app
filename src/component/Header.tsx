import React from 'react';
import {logoutData} from '../service/Action'
import { Grid, Button, AppBar, Toolbar, Link } from '@material-ui/core';

interface IHeaderProps {
    store:any,
    cityName:string,
    setCurrentLocation:(e:any) => void
}

interface IHeaderStates {
    userName:string
}

export class Header extends React.Component<IHeaderProps, IHeaderStates> {
    constructor(props: any) {
        super(props)
        this.onSubmitClick = this.onSubmitClick.bind(this);
        this.state = {userName: this.props.store.getState().userName}
    }

    onSubmitClick(e:any){
        this.props.store.dispatch(logoutData());
        location.reload();
    }

    render() {
        return (
                <Grid container alignItems={"center"} style={{marginBottom:'1rem'}}>
                    <Grid item style={{flexGrow:1}}>
                        <AppBar position={"static"} style={{flexGrow:1, flexDirection:'row'}}>
                            <Toolbar style={{flexGrow:1, flexDirection:'row'}}>
                                <Grid item id="welcomeUser" style={{flexGrow:1}} > Welcome {this.state.userName}</Grid>
                                <Grid item>{this.props.cityName}</Grid>
                                <Link style={{color:"white", cursor:"pointer", padding:'1rem'}} onClick={this.props.setCurrentLocation}><i>choose current location</i></Link>
                                <Grid item ><Button onClick={this.onSubmitClick} variant={"contained"} >Logout</Button></Grid>
                            </Toolbar>
                        </AppBar>
                    </Grid>
                </Grid> 
        )
    }
}

export default Header;;