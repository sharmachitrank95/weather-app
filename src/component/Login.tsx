import React from 'react';
import { Redirect } from 'react-router';
import {loginData,logoutData} from '../service/Action'
import { Grid, Input, Button } from '@material-ui/core';

interface ILoginProps {
    store:any,
}

interface ILoginStates {
    userName:string,
    password:string
}

function getHomeStyle() {
    return {
        padding:"1rem",
    }
}

export class Login extends React.Component<ILoginProps, ILoginStates> {
    constructor(props: any) {
        super(props)
        this.state = {userName:'',password:''};
        this.onSubmitClick = this.onSubmitClick.bind(this);
        this.handleNameChange = this.handleNameChange.bind(this);
        this.handlePasswordChange = this.handlePasswordChange.bind(this);
        this.renderRedirect = this.renderRedirect.bind(this);
    }

    renderRedirect(){
        if (!!localStorage.getItem('userName') && localStorage.getItem('userName')!.length > 0) {
          return <Redirect to={process.env.PUBLIC_URL +'/webApp'} />
        }
    }

    onSubmitClick(e:any){
        this.props.store.dispatch(loginData(this.state.userName));
        this.renderRedirect();
        location.reload();
    }

    handleNameChange(e:any){
        this.setState({
            userName: e.target.value
        });
    }

    handlePasswordChange(e:any){
        this.setState({
            password:e.target.value
        });
    }

    render() {
        return (
            <Grid container direction={"column"} alignItems={"center"} justify={"center"} style={{height:"100%"}}>
                {this.renderRedirect()}
                <Grid item style={getHomeStyle()} >
                    <Input id="userName" placeholder="name" required  autoComplete={'false'} type="name" onChange={this.handleNameChange}/>
                </Grid> 
                 <Grid item style={getHomeStyle()}>  
                    <Input placeholder="password"  type="password" />
                </Grid>
                <Grid item style={getHomeStyle()}>  
                    <Button id="loginButton" variant={"contained"} disabled = {this.state.userName.length == 0} onClick={this.onSubmitClick}>LogIn</Button>
                </Grid>
            </Grid>
        )
    }
}

export default Login;