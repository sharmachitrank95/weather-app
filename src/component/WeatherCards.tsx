import React from 'react';
import {Card, CardHeader, CardContent, Typography, Grid} from '@material-ui/core';

interface IWeatherCardsProps{
    WeatherCardData:[]
}
interface IWeatherCardsStates{

}

export class WeatherCards extends React.Component<IWeatherCardsProps,IWeatherCardsStates>{
    constructor(props:any) {
        super(props);
    }

    render(){
        let cards;
        cards = this.props.WeatherCardData.map((cardData:any)=>{
            return (
                <Grid item xs={"auto"}>
                    <Card>
                        <CardContent>
                            <Typography>
                               Time: {cardData.dt_txt.split(' ')[1]}
                            </Typography>
                            <Typography variant="h3">
                                {cardData.weather[0].description}
                            </Typography>
                            <Typography>
                                Max Temp: {cardData.main.temp_max}
                            </Typography>
                            <Typography>
                                Min Temp: {cardData.main.temp_min}
                            </Typography>
                            <Typography>
                                Pressure: {cardData.main.pressure}
                            </Typography>
                            <Typography>
                                Humidity: {cardData.main.humidity}
                            </Typography>
                        </CardContent>
                    </Card>
                </Grid>
            )
        })
        return(
            <Grid item xs={10}>
                <Grid container spacing={24} style={{flexGrow:1}}>
                   {cards}
                </Grid>
            </Grid>
        )
        
    }
}

export default WeatherCards;