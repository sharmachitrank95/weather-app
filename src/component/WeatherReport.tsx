import React from 'react';
import {DateNavigator} from '../component/DateNavigator';
import {WeatherCards} from '../component/WeatherCards';
import * as _ from 'lodash';
import { Grid } from '@material-ui/core';
import CircularProgress from '@material-ui/core/CircularProgress';

interface IWeatherReportProps{
    weatherData:any
}
interface IWeatherReportStates{
    date:any,
    cardsData:any,
    currentDate:any
}

export class WeatherReport extends React.Component<IWeatherReportProps,IWeatherReportStates>{
    public availableDates:Array<string>;
    constructor(props:any) {
        super(props);
        this.state = {cardsData:null,date:null,currentDate:null};
        this.availableDates = [];
        this.handleDateChange = this.handleDateChange.bind(this);
        this.setCardData = this.setCardData.bind(this);
        this.fetchAvailableDates();
    }

    fetchAvailableDates(){
        this.props.weatherData.list.map((element:any) => {
            element = element.dt_txt.split(' ')[0];
            if(this.availableDates.indexOf(element) === -1){
                this.availableDates.push(element);
            }
        })
    }
    componentWillMount(){
        this.setCardData(this.availableDates[0],this.props.weatherData);
    }

    handleDateChange(e:any,value:any){
       this.setCardData(value,this.props.weatherData);
    }

    setCardData(date:any,weatherData:any){
        let data = _.filter(weatherData.list,(element:any) =>{
           return element.dt_txt.split(' ')[0] === date;
        })
        this.setState({
            currentDate: date,
            cardsData:_.cloneDeep(data)
        })
    }

    render(){
        return(
            <Grid container spacing={8}>
                <DateNavigator dates={this.availableDates} handleDateChange={this.handleDateChange} currentDate={this.state.currentDate}/>
               {!!this.state.cardsData && <WeatherCards WeatherCardData={this.state.cardsData}/>}
            </Grid>
        )
    }
}

export default WeatherReport;

