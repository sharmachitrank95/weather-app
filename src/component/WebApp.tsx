import React from 'react';
import { Redirect } from 'react-router';
import {Header} from '../component/Header';
import {WeatherReport} from '../component/WeatherReport';
import * as _ from 'lodash';
import { CircularProgress } from '@material-ui/core';

interface IWebAppStates{
    xCoordinate: Number,
    yCoordinate:Number,
    weatherData:any,
    isBusy:boolean
}
interface IWebAppProps{
    store:any
}

export class WebApp extends React.Component<IWebAppProps, IWebAppStates>{
    private cityName:string = '';
    constructor(props:any){
        super(props);
        this.state = {xCoordinate:30.734301,yCoordinate:76.793297,weatherData:null,isBusy:false};
        this.setCurrentLocation = this.setCurrentLocation.bind(this);
        this.renderRedirect = this.renderRedirect.bind(this);
    }

    setCurrentLocation(){
        navigator.geolocation.getCurrentPosition((position)=>{
            if(position){
                this.setState({
                    xCoordinate:position.coords.latitude,
                    yCoordinate:position.coords.longitude
                })
            }
            this.getData();
        });
    }

    shouldComponentUpdate(nextProps:any,nextState:any){
        if(!this.state.weatherData || (!!this.state.weatherData && this.state.weatherData.city.name !== nextState.weatherData.city.name)){
            return true;
        }
        if(this.state.isBusy !== nextState.isBusy){
            return true;
        }
        return false;
    }
    
    componentDidMount(){
        this.getData();
    }

    getData(){
        this.setState({
             isBusy:true
        })
        fetch('https://cors-anywhere.herokuapp.com/api.openweathermap.org/data/2.5/forecast?lat='+ this.state.xCoordinate +'&lon='+ this.state.yCoordinate +'&appid=0cefc283fa0c3ad68584e2c30b643c94',
        {
            headers:{'Content-Type': 'text/plain','Access-Type':'application/json'},
        })
        .then((data:any)=>{
           return data.json();
        })
        .then((data)=>{
            this.cityName = data.city.name;
            this.setState({
                isBusy:false,
                weatherData:_.cloneDeep(data) 
            })
        })
    }
    renderRedirect(){
        if (!localStorage.getItem('userName') || localStorage.getItem('userName')!.length == 0) {
          return <Redirect to={process.env.PUBLIC_URL +'/login'} />
        }
    }

    render(){
        let weatherReport;
        let spinner;
        weatherReport = <div id="weatherReport">
            <Header store={this.props.store} cityName={this.cityName} setCurrentLocation={this.setCurrentLocation}/>
            {!!this.state.weatherData && <WeatherReport weatherData={this.state.weatherData}/>}
        </div>
        spinner = <CircularProgress style={{position:'fixed',top:'50%',right:'50%',left:'50%',bottom:'50%'}} size={100}></CircularProgress>
        return(
            <React.Fragment>
                {this.renderRedirect()}
                {this.state.isBusy? spinner: weatherReport}
            </React.Fragment> 
        )
    }
}

export default WebApp;
