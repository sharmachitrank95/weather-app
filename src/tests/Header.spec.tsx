import { configure, shallow, mount } from 'enzyme';
import {Header} from '../component/Header'
import {createStore} from 'redux';
import myApp from '../service/Reducer';
import * as React from 'react';
import Adapter from 'enzyme-react-adapter-future';

configure({ adapter: new Adapter() });

const store = createStore(myApp);
const mockSetLocation = (e:any)=>{}

describe('<Login />',()=>{
    it('should render correctly',()=>{
        const loginComponent = <Header store={store} cityName={"chandigarh"} setCurrentLocation={mockSetLocation} />;
        const wrapper = shallow(loginComponent);
        expect(wrapper).toMatchSnapshot();   
    });

    // it('should render country name correctly',()=>{
    //     store.dispatch(loginData("Chitrank"));
    // })
})