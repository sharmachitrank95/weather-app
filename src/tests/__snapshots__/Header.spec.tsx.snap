// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    cityName="chandigarh"
    setCurrentLocation={[Function]}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "alignItems": "center",
      "children": <WithStyles(Grid)
        item={true}
        style={
          Object {
            "flexGrow": 1,
          }
        }
      >
        <WithStyles(AppBar)
          position="static"
          style={
            Object {
              "flexDirection": "row",
              "flexGrow": 1,
            }
          }
        >
          <WithStyles(Toolbar)
            style={
              Object {
                "flexDirection": "row",
                "flexGrow": 1,
              }
            }
          >
            <WithStyles(Grid)
              id="welcomeUser"
              item={true}
              style={
                Object {
                  "flexGrow": 1,
                }
              }
            >
               Welcome 
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
            >
              chandigarh
            </WithStyles(Grid)>
            <WithStyles(Link)
              onClick={[Function]}
              style={
                Object {
                  "color": "white",
                  "cursor": "pointer",
                  "padding": "1rem",
                }
              }
            >
              <i>
                choose current location
              </i>
            </WithStyles(Link)>
            <WithStyles(Grid)
              item={true}
            >
              <WithStyles(Button)
                onClick={[Function]}
                variant="contained"
              >
                Logout
              </WithStyles(Button)>
            </WithStyles(Grid)>
          </WithStyles(Toolbar)>
        </WithStyles(AppBar)>
      </WithStyles(Grid)>,
      "container": true,
      "style": Object {
        "marginBottom": "1rem",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(AppBar)
          position="static"
          style={
            Object {
              "flexDirection": "row",
              "flexGrow": 1,
            }
          }
        >
          <WithStyles(Toolbar)
            style={
              Object {
                "flexDirection": "row",
                "flexGrow": 1,
              }
            }
          >
            <WithStyles(Grid)
              id="welcomeUser"
              item={true}
              style={
                Object {
                  "flexGrow": 1,
                }
              }
            >
               Welcome 
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
            >
              chandigarh
            </WithStyles(Grid)>
            <WithStyles(Link)
              onClick={[Function]}
              style={
                Object {
                  "color": "white",
                  "cursor": "pointer",
                  "padding": "1rem",
                }
              }
            >
              <i>
                choose current location
              </i>
            </WithStyles(Link)>
            <WithStyles(Grid)
              item={true}
            >
              <WithStyles(Button)
                onClick={[Function]}
                variant="contained"
              >
                Logout
              </WithStyles(Button)>
            </WithStyles(Grid)>
          </WithStyles(Toolbar)>
        </WithStyles(AppBar)>,
        "item": true,
        "style": Object {
          "flexGrow": 1,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Toolbar)
            style={
              Object {
                "flexDirection": "row",
                "flexGrow": 1,
              }
            }
          >
            <WithStyles(Grid)
              id="welcomeUser"
              item={true}
              style={
                Object {
                  "flexGrow": 1,
                }
              }
            >
               Welcome 
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
            >
              chandigarh
            </WithStyles(Grid)>
            <WithStyles(Link)
              onClick={[Function]}
              style={
                Object {
                  "color": "white",
                  "cursor": "pointer",
                  "padding": "1rem",
                }
              }
            >
              <i>
                choose current location
              </i>
            </WithStyles(Link)>
            <WithStyles(Grid)
              item={true}
            >
              <WithStyles(Button)
                onClick={[Function]}
                variant="contained"
              >
                Logout
              </WithStyles(Button)>
            </WithStyles(Grid)>
          </WithStyles(Toolbar)>,
          "position": "static",
          "style": Object {
            "flexDirection": "row",
            "flexGrow": 1,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Grid)
                id="welcomeUser"
                item={true}
                style={
                  Object {
                    "flexGrow": 1,
                  }
                }
              >
                 Welcome 
              </WithStyles(Grid)>,
              <WithStyles(Grid)
                item={true}
              >
                chandigarh
              </WithStyles(Grid)>,
              <WithStyles(Link)
                onClick={[Function]}
                style={
                  Object {
                    "color": "white",
                    "cursor": "pointer",
                    "padding": "1rem",
                  }
                }
              >
                <i>
                  choose current location
                </i>
              </WithStyles(Link)>,
              <WithStyles(Grid)
                item={true}
              >
                <WithStyles(Button)
                  onClick={[Function]}
                  variant="contained"
                >
                  Logout
                </WithStyles(Button)>
              </WithStyles(Grid)>,
            ],
            "style": Object {
              "flexDirection": "row",
              "flexGrow": 1,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  " Welcome ",
                  null,
                ],
                "id": "welcomeUser",
                "item": true,
                "style": Object {
                  "flexGrow": 1,
                },
              },
              "ref": null,
              "rendered": Array [
                " Welcome ",
                null,
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "chandigarh",
                "item": true,
              },
              "ref": null,
              "rendered": "chandigarh",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <i>
                  choose current location
                </i>,
                "onClick": [Function],
                "style": Object {
                  "color": "white",
                  "cursor": "pointer",
                  "padding": "1rem",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "choose current location",
                },
                "ref": null,
                "rendered": "choose current location",
                "type": "i",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(Button)
                  onClick={[Function]}
                  variant="contained"
                >
                  Logout
                </WithStyles(Button)>,
                "item": true,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Logout",
                  "onClick": [Function],
                  "variant": "contained",
                },
                "ref": null,
                "rendered": "Logout",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "alignItems": "center",
        "children": <WithStyles(Grid)
          item={true}
          style={
            Object {
              "flexGrow": 1,
            }
          }
        >
          <WithStyles(AppBar)
            position="static"
            style={
              Object {
                "flexDirection": "row",
                "flexGrow": 1,
              }
            }
          >
            <WithStyles(Toolbar)
              style={
                Object {
                  "flexDirection": "row",
                  "flexGrow": 1,
                }
              }
            >
              <WithStyles(Grid)
                id="welcomeUser"
                item={true}
                style={
                  Object {
                    "flexGrow": 1,
                  }
                }
              >
                 Welcome 
              </WithStyles(Grid)>
              <WithStyles(Grid)
                item={true}
              >
                chandigarh
              </WithStyles(Grid)>
              <WithStyles(Link)
                onClick={[Function]}
                style={
                  Object {
                    "color": "white",
                    "cursor": "pointer",
                    "padding": "1rem",
                  }
                }
              >
                <i>
                  choose current location
                </i>
              </WithStyles(Link)>
              <WithStyles(Grid)
                item={true}
              >
                <WithStyles(Button)
                  onClick={[Function]}
                  variant="contained"
                >
                  Logout
                </WithStyles(Button)>
              </WithStyles(Grid)>
            </WithStyles(Toolbar)>
          </WithStyles(AppBar)>
        </WithStyles(Grid)>,
        "container": true,
        "style": Object {
          "marginBottom": "1rem",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(AppBar)
            position="static"
            style={
              Object {
                "flexDirection": "row",
                "flexGrow": 1,
              }
            }
          >
            <WithStyles(Toolbar)
              style={
                Object {
                  "flexDirection": "row",
                  "flexGrow": 1,
                }
              }
            >
              <WithStyles(Grid)
                id="welcomeUser"
                item={true}
                style={
                  Object {
                    "flexGrow": 1,
                  }
                }
              >
                 Welcome 
              </WithStyles(Grid)>
              <WithStyles(Grid)
                item={true}
              >
                chandigarh
              </WithStyles(Grid)>
              <WithStyles(Link)
                onClick={[Function]}
                style={
                  Object {
                    "color": "white",
                    "cursor": "pointer",
                    "padding": "1rem",
                  }
                }
              >
                <i>
                  choose current location
                </i>
              </WithStyles(Link)>
              <WithStyles(Grid)
                item={true}
              >
                <WithStyles(Button)
                  onClick={[Function]}
                  variant="contained"
                >
                  Logout
                </WithStyles(Button)>
              </WithStyles(Grid)>
            </WithStyles(Toolbar)>
          </WithStyles(AppBar)>,
          "item": true,
          "style": Object {
            "flexGrow": 1,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Toolbar)
              style={
                Object {
                  "flexDirection": "row",
                  "flexGrow": 1,
                }
              }
            >
              <WithStyles(Grid)
                id="welcomeUser"
                item={true}
                style={
                  Object {
                    "flexGrow": 1,
                  }
                }
              >
                 Welcome 
              </WithStyles(Grid)>
              <WithStyles(Grid)
                item={true}
              >
                chandigarh
              </WithStyles(Grid)>
              <WithStyles(Link)
                onClick={[Function]}
                style={
                  Object {
                    "color": "white",
                    "cursor": "pointer",
                    "padding": "1rem",
                  }
                }
              >
                <i>
                  choose current location
                </i>
              </WithStyles(Link)>
              <WithStyles(Grid)
                item={true}
              >
                <WithStyles(Button)
                  onClick={[Function]}
                  variant="contained"
                >
                  Logout
                </WithStyles(Button)>
              </WithStyles(Grid)>
            </WithStyles(Toolbar)>,
            "position": "static",
            "style": Object {
              "flexDirection": "row",
              "flexGrow": 1,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(Grid)
                  id="welcomeUser"
                  item={true}
                  style={
                    Object {
                      "flexGrow": 1,
                    }
                  }
                >
                   Welcome 
                </WithStyles(Grid)>,
                <WithStyles(Grid)
                  item={true}
                >
                  chandigarh
                </WithStyles(Grid)>,
                <WithStyles(Link)
                  onClick={[Function]}
                  style={
                    Object {
                      "color": "white",
                      "cursor": "pointer",
                      "padding": "1rem",
                    }
                  }
                >
                  <i>
                    choose current location
                  </i>
                </WithStyles(Link)>,
                <WithStyles(Grid)
                  item={true}
                >
                  <WithStyles(Button)
                    onClick={[Function]}
                    variant="contained"
                  >
                    Logout
                  </WithStyles(Button)>
                </WithStyles(Grid)>,
              ],
              "style": Object {
                "flexDirection": "row",
                "flexGrow": 1,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    " Welcome ",
                    null,
                  ],
                  "id": "welcomeUser",
                  "item": true,
                  "style": Object {
                    "flexGrow": 1,
                  },
                },
                "ref": null,
                "rendered": Array [
                  " Welcome ",
                  null,
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "chandigarh",
                  "item": true,
                },
                "ref": null,
                "rendered": "chandigarh",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <i>
                    choose current location
                  </i>,
                  "onClick": [Function],
                  "style": Object {
                    "color": "white",
                    "cursor": "pointer",
                    "padding": "1rem",
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "choose current location",
                  },
                  "ref": null,
                  "rendered": "choose current location",
                  "type": "i",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(Button)
                    onClick={[Function]}
                    variant="contained"
                  >
                    Logout
                  </WithStyles(Button)>,
                  "item": true,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Logout",
                    "onClick": [Function],
                    "variant": "contained",
                  },
                  "ref": null,
                  "rendered": "Logout",
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
